### PR TITLE
chore(ci): add issue labeling reporter 

### LINKED ISSUE #16273: Run a GitHub Action weekly to check for issues that are missing labels
## Intro

I wrote some GitHub issue search queries that help us check if any issues are wrongly labeled.

Maybe we can automate this task, by writing a GitHub Action script.

## Requirements

1. Action runs weekly
1. Check if any queries have a match
    - If a query matches (so an issue is missing labels), throw error
    - If no query matches, exit cleanly

### Queries

```markdown
https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+-label%3Astatus%3Arequirements+-label%3Astatus%3Aready+-label%3Astatus%3Ain-progress+-label%3Astatus%3Ablocked+-label%3Astatus%3Awaiting-on-response+

https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+-label%3Atype%3Abug+label%3Astatus%3Aready+-label%3Atype%3Afeature+-label%3Atype%3Adocs+-label%3Atype%3Arefactor+

https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+-label%3Apriority-1-critical+-label%3Apriority-2-high+-label%3Apriority-3-medium+-label%3Apriority-4-low++-label%3Apriority-5-triage
```

## Todo:

- [ ] Put "type of issue" query in action
- [ ] Put "priority query" in action
- [ ] Run action weekly
- [ ] Make action error if any query matches
- [ ] Else action exits cleanly
- [ ] If action works properly, update [`docs/development/issue-labeling`](https://github.com/renovatebot/renovate/blob/main/docs/development/issue-labeling.md) to mention the GitHub action check run

### PR DESCRIPTION
**Pull request by denis-rossati**
## Changes

This PR will create a new workflow that once in a week will check all open issues and search if any of them are not labeled correctly. If all of them are labeled as expected, it'll cleanly complete the job, otherwise, a new issue reporting the unlabeled issues will be created.

Currently the new issue does not have an label, though. I don't know if we should create a particular label just for that kind of issue or if the action should ignore them.

## Context

Explained at #16273 

## Related issues

- Closes #16273

## Documentation

- [x] I have updated the documentation.

## How I've tested my work

- [x] Tested the new actions in a real repository and using [act](https://github.com/nektos/act).

### CODE REVIEW

File: .github/workflows/check-unlabeled-issues.yml
Comment (viceice) [inline .github/workflows/check-unlabeled-issues.yml:12]: it should use the build-in token. also set workflow permissions explicit

File: .github/workflows/check-unlabeled-issues.yml
Comment (viceice) [inline .github/workflows/check-unlabeled-issues.yml:53]: we should only create one issue and reuse it (reopen, comment).

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [inline .github/workflows/check-unlabeled-issues.yml:4]:
```suggestion
    # Run every Sunday at midnight
```

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [reply .github/workflows/check-unlabeled-issues.yml:12]: > also set workflow permissions explicit

The [GitHub Docs, assigning permissions to jobs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) has more information about this.

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [inline .github/workflows/check-unlabeled-issues.yml:17]:
```suggestion
          # When the repository labels are changed (i.e dropped a label, added a label, etc), you should make the same change to the lists below.
```

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [inline .github/workflows/check-unlabeled-issues.yml:18]:
```suggestion
          # For example: if the repository added a "to-do" status label, then add "-label:status:to-do" to the STATUS_LABELS_FILTER.
```

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [inline .github/workflows/check-unlabeled-issues.yml:39]:
```suggestion
              # Format the output to be a list of issue numbers
```

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [inline .github/workflows/check-unlabeled-issues.yml:58]:
```suggestion
            # Fail the action to if there are issues missing the correct labels
```

File: docs/development/issue-labeling.md
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:227]:
```suggestion
We have a GitHub Action that checks if we're using the correct labels on our issues.
The action runs once in a week and checks for issues without the proper labels.
```

File: docs/development/issue-labeling.md
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:229]:
```suggestion
If the action finds any issues missing labels, it creates a new issue with the missing labels and details the issues found so any collaborator can label them accordingly.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Some small changes to start with.

After the code is changed to implement @viceice's comments/suggestions, we should run the shellcheck program on the bash parts of the code, to make sure it's safe.

https://www.shellcheck.net/

**Comment by viceice** [comment]
> Some small changes to start with.
> 
> After the code is changed to implement @viceice's comments/suggestions, we should run the shellcheck program on the bash parts of the code, to make sure it's safe.
> 
> https://www.shellcheck.net/

That code should probably inside a `.sh` file inside `tools` folder, so it can be easily validated via shellcheck

**Comment by HonkingGoose** [comment]
> That code should probably inside a `.sh` file inside `tools` folder, so it can be easily validated via shellcheck

That's way better, as we run shellcheck automatically, I think. So we know the file is good now, and later. 😄

**Comment by denis-rossati** [comment]
> > That code should probably inside a `.sh` file inside `tools` folder, so it can be easily validated via shellcheck
> 
> That's way better, as we run shellcheck automatically, I think. So we know the file is good now, and later. 😄

That's a good idea, should the `.sh` file be in the `.github` folder or other place?

File: .github/workflows/check-unlabeled-issues.yml
Comment (HonkingGoose) [reply .github/workflows/check-unlabeled-issues.yml:53]: We have another GitHub action that _locks_ closed issues after x amount of days.

Would the "label check" action be able to re-open the issue? I suspect we need to write a special  rule somewhere so the lock action stays away from the "missing label "-issue.

Comment (denis-rossati) [reply .github/workflows/check-unlabeled-issues.yml:53]: It seems that the lock is for issues that have been inactive for more than 30 days, which means that after about the 4th check it will block the issue

From what I've seen in the lock-threads, there is [this condition](https://github.com/dessant/lock-threads#:~:text=exclude%2Dany%2Dissue,defaults%20to%20%27%27) which excludes any issue with a particular label from being locked (maybe something like an `action-label-check` label?), this would also help to find and edit the issue instead of creating a new one. I just don't know if there's a way to prevent other contributors from deliberately using this label to prevent their issues from being locked

Comment (viceice) [reply .github/workflows/check-unlabeled-issues.yml]: a normal contributer can't change labels

Comment (HonkingGoose) [reply .github/workflows/check-unlabeled-issues.yml]: To apply/dismiss labels you need at least triage level access to the repository.

So normal users can not abuse the special label to prevent their issue from getting closed.

Comment (denis-rossati) [reply .github/workflows/check-unlabeled-issues.yml]: Oh great, so it's ok to use this approach?

Comment (HonkingGoose) [reply .github/workflows/check-unlabeled-issues.yml]: I think these are the next steps:

- update lock action so it ignores the label for the "label-issues"-issue
- let the "label issues" action re-open closed issues when needed
- put the Bash code in a separate `.sh` file like @viceice said

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: tools/check-unlabeled-issues.sh
Comment (viceice) [inline tools/check-unlabeled-issues.sh:5]: as already stated: status isn't a required label

Comment (denis-rossati) [reply tools/check-unlabeled-issues.sh:5]: Thanks, I didn't noticed I left one behind

**Review by HonkingGoose — COMMENTED** [review]
I'll let the maintainers review the shell script and GitHub Actions code and permissions.

I'm happy with the docs.

File: tools/check-unlabeled-issues.sh
Comment (rarkins) [inline tools/check-unlabeled-issues.sh:7]:
```suggestion
PRIORITY_LABELS_FILTER='-label:priority-1-critical -label:priority-2-high -label:priority-3-medium -label:priority-4-low'
```

File: tools/check-unlabeled-issues.sh
Comment (rarkins) [inline tools/check-unlabeled-issues.sh:13]:
```suggestion
for FILTER in "$TYPE_LABELS_FILTER" "$PRIORITY_LABELS_FILTER"; do
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Seems this needs extra testing. How can it be done without accidentally "spamming" ourselves?

File: tools/check-unlabeled-issues.sh
Comment (viceice) [inline tools/check-unlabeled-issues.sh:52]: log as error

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message

**Comment by rarkins** [comment]
Closing due to inactivity, reopening/resuming later is welcome

**Pull request closed without merging**
